src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+(() => {
+  const body = document.querySelector('body');
+
+  let left = false;
+  let right = false;
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+  });
+
+  const notification = (parentNode, classes, text) => {
+    const node = document.createElement('div');
+
+    node.classList.add(...classes);
+    node.textContent = text;
+    node.dataset.qa = 'notification';
+    parentNode.append(node);
+  };
+
+  const firstPromise = new Promise((resolve, reject) => {
+    document.addEventListener('click', () => {
+      resolve('First promise was resolved');
+      left = true;
+    });
+
+    setTimeout(() => {
+      if (!left) {
+        reject(new Error('First promise was rejected'));
+      }
+    }, 3000);
```

Comment:
You have a race condition here. The `setTimeout` function will execute after 3 seconds regardless of whether the 'click' event has been fired or not. This means that if the user clicks after 3 seconds, the promise will be resolved, but the `setTimeout` function will still run and reject the promise. To fix this, you can use `clearTimeout` inside the 'click' event listener to clear the timeout when the user clicks.